### DISCUSSION TITLE
WindFind 1.1.0

### DISCUSSION THREAD

**Discussion by kingToolbox** [kingToolbox/WindFind#4] [Announcements]
# Spotlight

## Any user, any file system

`WindFind` supports **any user** indexing **any file system**, as long as the operating system can read it, everything can be indexed quickly.

## Super-fast speed, ultra-small index

**`WindFind`'s folder traversal speed is unmatched**. Once it detects administrator privileges, it directly indexes the `MFT`, delivering incredibly fast performance. The thoughtfully designed index data structure allows each file to be represented with just **4 bytes**, capturing its path, file size, and modification time, resulting in **an index file that is smaller than ever before**.

## Rich features that meet your every need.

More than just indexing and searching—**it's a master at displaying file lists and calculating sizes**! Enjoy vibrant colors, striking icons, and lightning-fast printing speeds, all designed to elevate your file browsing experience to new heights!

# Usage:

### Systems

|   Version | Architecture |  Operating Systems |
| --------- | -------------- | --------------------- |
| WindFind_1.1.0_Windows_Portable_x86_64 | x86_64 | Windows 7, Windows 8, Windows 10, Windows 11 |
| WindFind_1.1.0_Linux_Portable_x86_64 | x86_64 | Generic Linux (glibc 2.17 and later) |

# Features:
- [APP] **Added Linux version support.**
- [INDEXER] **Supports configuring included/excluded directories during indexing.**
- [INDEXER] Supports more systems modifying parent directory timestamps on file changes, including `ext4`, `btrfs` and others. 
- [LISTER] Added automatic color inversion for executable names for better highlighting.

# Improvements:
- [CALCULATOR] **Significantly improved folder size calculation performance, with gains exceeding 50%.**
- [LISTER] Improved colors in the `dige` theme for better visibility on white backgrounds.
- [LISTER] Refined icons in the `dige` theme for clearer representation.

# Bugs:
- [INDEXER] Fixed the loss of some files when updating the index generated by the last scan using administrator privileges on Windows.
- [INDEXER] Resolved index clearing error caused by duplicate subfolder names in the same directory, particularly in virtual file systems.
- [INDEXER] Resolved inability to create disk index in Windows `Sandbox` with administrator permissions.

<hr /><em>This discussion was created from the release <a href='https://github.com/kingToolbox/WindFind/releases/tag/1.1.0'>WindFind 1.1.0</a>.</em>

**Comment by gooker** [comment]
It might be helpful to use some images to illustrate the basic usage process or methods.  like windterm Introduction 
 https://kingtoolbox.github.io/

**Reply by kingToolbox** [reply]
Yes, having images would make it more intuitive. When I was writing the `README.md`, the development hadn't been completed yet, so there were no images included. Recently, I've been busy developing the `WindFind GUI` version and reviewing the `WindTerm` source code to continue working on the remaining tasks for the `2.7.0` version. I will add screenshots as soon as possible. Thank you very much for your suggestion.

**Reply by Nyar233** [reply]
> Yes, having images would make it more intuitive. When I was writing the `README.md`, the development hadn't been completed yet, so there were no images included. Recently, I've been busy developing the `WindFind GUI` version and reviewing the `WindTerm` source code to continue working on the remaining tasks for the `2.7.0` version. I will add screenshots as soon as possible. Thank you very much for your suggestion.

Will the WindTerm 2.7.0 version be released in the next few months?

**Reply by kingToolbox** [reply]
@Nyar233 
Yes, this month!

I expect to release version `WindTerm 2.7.0 Prerelease 2` this week as a first step. In addition, version `WindTerm 2.8.0` is also nearly complete, so it should follow shortly after the release of version `WindTerm 2.7.0`. The majority of the development work was actually completed back in July. It's just that the temporary development of `WindFind` and frequent travel have caused the delay until now.

Thank you for your patience.

**Reply by Nyar233** [reply]
> @Nyar233 Yes, this month!
> 
> I expect to release version `WindTerm 2.7.0 Prerelease 2` this week as a first step. In addition, version `WindTerm 2.8.0` is also nearly complete, so it should follow shortly after the release of version `WindTerm 2.7.0`. The majority of the development work was actually completed back in July. It's just that the temporary development of `WindFind` and frequent travel have caused the delay until now.
> 
> Thank you for your patience.

Thank you for your detailed response. Your work is worth waiting for, because Windterm is such a great product. The arrival of the new product, WindFind, is very exciting, and we look forward to its growth and success.
